src/datumaro/experimental/categories.py

```diff
@@ -131,6 +134,231 @@ def __hash__(self):
+        if index is not None:
+            return index, self.items[index]
+        return None, None
+
+    def get_children(self, parent_name: str) -> Tuple[str, ...]:
+        """
+        Get all children of a parent label.
+
+        Args:
+            parent_name: The name of the parent label
+
+        Returns:
+            Tuple of child label names
+        """
+        return self._children_map.get(parent_name, ())
+
+    def get_parent(self, label_name: str) -> Optional[str]:
+        """
+        Get the parent of a label.
+
+        Args:
+            label_name: The name of the label
+
+        Returns:
+            Parent name or None if no parent
+        """
+        index = self._index_map.get(label_name)
+        if index is not None:
+            return self.items[index].parent
+        return None
+
+    def get_hierarchy_level(self, label_name: str) -> int:
+        """
+        Get the hierarchy level of a label (0 for root, 1 for first level children, etc.)
+
+        Args:
+            label_name: The name of the label
+
+        Returns:
+            Hierarchy level
+        """
+        level = 0
+        current = label_name
+        while True:
+            parent = self.get_parent(current)
+            if not parent:
+                break
+            level += 1
+            current = parent
+        return level
+
+    def __getitem__(self, idx: int) -> HierarchicalLabelCategory:
+        """Get category by index."""
+        return self.items[idx]
+
+    def __contains__(self, value: Union[int, str]) -> bool:
+        """Check if a label exists by name or index."""
+        if isinstance(value, str):
+            return value in self._index_map
+        else:
+            return 0 <= value < len(self.items)
+
+    def __len__(self) -> int:
+        """Get the number of labels."""
+        return len(self.items)
+
+    def __iter__(self) -> Iterator[HierarchicalLabelCategory]:
+        """Iterate over label categories."""
+        return iter(self.items)
+
+    def __hash__(self):
+        # Hash label_groups via value-based representation to avoid relying on their own hash implementation.
+        lg_repr = tuple((lg.name, tuple(lg.labels), lg.group_type) for lg in self.label_groups)
+        return hash((self.items, lg_repr, frozenset(self.label_semantics.items())))
+
+    @classmethod
+    def from_iterable(
+        cls,
+        iterable: Iterable[
+            Union[
+                str,
+                Tuple[str],
+                Tuple[str, str],
+                Tuple[str, str, Dict[str, Any]],
+            ]
+        ],
+    ) -> "HierarchicalLabelCategories":
+        """
+        Creates a HierarchicalLabelCategories from iterable.
+
+        Args:
+            iterable: This iterable object can be:
+                - a list of str - will be interpreted as list of Category names
+                - a list of positional arguments - will generate Categories with these arguments
```

Comment:
I think an example will help to understand this method.

Reply:
I've removed this method, it shouldn't really be used and was copied over from the old datumaro logic.